app/front/scripts/services/history.js

```diff
@@ -0,0 +1,47 @@
+/**
```

Comment:
Why have we opted for this custom history handling instead of saving the state using the URL?